### PR TITLE
testcne:add new ring API test

### PR DESCRIPTION
**Pull request by KeithWiles**
Add a test for the new cne_ring_init() API and verify it returns errors or valid
ring entries with the new arguments.

Signed-off-by: Keith Wiles <keith.wiles@intel.com>

### CODE REVIEW

File: test/testcne/ring_test.c [hunk 1]
```diff
@@ -108,6 +114,97 @@ ring_name_tests(void *args)
...
+    tst_ok("Using ring_size=%zu\n", ring_size);
+
+    tsize = cne_ring_get_memsize_elem(0, ring_size);
+    TST_ASSERT_AND_CLEANUP(tsize > 0, "Ring memsize < 0\n", test_ring_cleanup, &tst);
+
+    tst.mm = mmap_alloc((tsize / CNE_CACHE_LINE_SIZE), CNE_CACHE_LINE_SIZE, MMAP_HUGEPAGE_4KB);
+    TST_ASSERT_AND_CLEANUP(tst.mm != NULL, "Out of memory\n", test_ring_cleanup, &tst);
+
+    sz = mmap_size(tst.mm, &bufcnt, &bufsz);
+    CNE_INFO("tsize: %ld, mmap size %ld, bufcnt %d, bufsz %d\n", tsize, sz, bufcnt, bufsz);
```
Comment (elzamath) [inline test/testcne/ring_test.c:139]: tsize - %zd
sz - %zu
bufcnt, bufsz - %u
would these format specifiers be better?

File: test/testcne/ring_test.c [hunk 2]
```diff
@@ -108,6 +114,97 @@ ring_name_tests(void *args)
...
+
+    TST_ASSERT_AND_CLEANUP(sz >= ring_size, "mmap size too small\n", test_ring_cleanup, &tst);
+
+    tst.r = cne_ring_init(mmap_addr(tst.mm), 0, "ring init", 8, ring_size, 0);
+    TST_ASSERT_AND_CLEANUP(tst.r == NULL, "Ring invalid size should have failed\n",
+                           test_ring_cleanup, &tst);
+
+    tst.r = cne_ring_init(mmap_addr(tst.mm), sz - (4 * 1024), "ring init", 8, ring_size, 0);
```
Comment (elzamath) [inline test/testcne/ring_test.c:147]: can 1024 be replaced with RING_SIZE macro instead?

File: test/testcne/ring_test.c [hunk 3]
```diff
@@ -108,6 +114,97 @@ ring_name_tests(void *args)
...
+    TST_ASSERT_AND_CLEANUP(ring_size == cne_ring_get_size(tst.r),
+                           "ring size=%d expected instead of %d\n", test_ring_cleanup, &tst,
+                           ring_size - 1, cne_ring_get_size(tst.r));
+
+    for (i = 0; i < ring_size / 2; i++)
+        TST_ASSERT_AND_CLEANUP(0 == cne_ring_enqueue(tst.r, (void *)((uintptr_t)i + 1)),
+                               "Enqueue i=%u", test_ring_cleanup, &tst, i + 1);
+
+    TST_ASSERT_AND_CLEANUP(i == cne_ring_count(tst.r),
+                           "Ring count=%d mismatch with enqueued=%d elements\n", test_ring_cleanup,
```
Comment (elzamath) [inline test/testcne/ring_test.c:179]: since ring_size and i are both unsigned ints, should we replace %d with %u when ring_size and i are being printed?

**Review by elzamath — APPROVED** [review]

File: test/testcne/ring_test.c (hunk 3 above)
Comment (KeithWiles) [reply test/testcne/ring_test.c:179]: I did not change this code in my patch, but I will update it.

**Pull request merged**